packager/react-packager/src/DependencyResolver/fastfs.js

```diff
@@ -293,16 +293,16 @@ class File {
   }
 
   getFiles() {
-    const files = [];
-    Object.keys(this.children).forEach(key => {
+    const files = Object.keys(this.children).map(key => {
       const file = this.children[key];
       if (file.isDir) {
-        files.push(...file.getFiles());
-      } else {
-        files.push(file);
+        return file.getFiles();
```

Comment:
Yoosa want to shave yakyak? Meesa shave yakyak...

(A) the `[].concat(...files)` approach:
```
getAllFiles() took 116 ms
getAllFiles() took 124 ms
getAllFiles() took 149 ms
getAllFiles() took 123 ms
getAllFiles() took 138 ms
getAllFiles() took 162 ms
```

(B) the `files = files.concat(file.getFiles())` approach:
```
getAllFiles() took 71 ms
getAllFiles() took 85 ms
getAllFiles() took 69 ms
getAllFiles() took 74 ms
getAllFiles() took 73 ms
getAllFiles() took 68 ms
```

(C) the `forEach(file => files.push(file));` approach
```
getAllFiles() took 100 ms
getAllFiles() took 93 ms
getAllFiles() took 99 ms
getAllFiles() took 102 ms
getAllFiles() took 106 ms
getAllFiles() took 100 ms
```

So, (B) turned out to the fastest, despite looking like O(n^2) on the surface, as @ide pointed out. I'll go back to (B) then so we can save literally **dozens** of milliseconds.